src/MongoDB.Driver/Linq/Translators/AggregateLanguageTranslator.cs

```diff
@@ -1276,5 +1303,57 @@ private bool TryTranslateSumResultOperator(PipelineExpression node, out BsonValu
             result = null;
             return false;
         }
+
+        private static class TemplateHelper
+        {
+            private const string MapTemplateStartDelimiter = "#{{";
+            private const string MapTemplateEndDelimiter = "}}#";
+            private const string MapTemplate = MapTemplateStartDelimiter + "{{{0}}}" + MapTemplateEndDelimiter;
```

Comment:
>What if a document just happens to have a field name that "looks" like a template to our code?

I expect that document cannot have a field name with `#{` or '}#' symbols since `PredicateTranslator` generates query based on c# linq expression which doesn't allow these field names.

>than having a class representing a template?

I'm not sure what you are suggesting. Do you mean creation some json template (with class representation) which will be added to translated value instead of `MapTemplate`. 
Having something like this:
        
        JsonTemplate.Create(memberInfo.OutOfCurrentScopePrefix) + inValue.ToString().Substring(...

Instead of:
        
        TemplateHelper.CreateTemplate(memberInfo.OutOfCurrentScopePrefix) + inValue.ToString().Substring(...

If so, how do you suggest to replace template values after translation?

Reply:
I don't actually have an alternate suggestion. But I am expressing the feedback that specially formatted strings feel like a potentially dangerous hack.